.hlint-universum.yaml

```diff
@@ -0,0 +1,1050 @@
+
+- hint: {lhs: "foldl' (flip f)", rhs: "flipfoldl' f"}
+
+- warn: {lhs: f >>= guard, rhs: guardM}
+- warn: {lhs: guard =<< f, rhs: guardM}
+
+- hint: {lhs: fmap (fmap f) x, rhs: f <<$>> x}
+
+- warn: {lhs: fmap concat (mapM f s), rhs: Universum.concatMapM f s}
+- warn: {lhs: concat <$> mapM f s, rhs: Universum.concatMapM f s}
+
+- warn: {lhs: fmap concat (forM f s), rhs: Universum.concatForM s f}
+- warn: {lhs: fmap concat (for f s), rhs: Universum.concatForM s f}
+- warn: {lhs: concat <$> forM f s, rhs: Universum.concatForM s f}
+- warn: {lhs: concat <$> for f s, rhs: Universum.concatForM s f}
+
+- hint: { lhs: fmap and (sequence s), rhs: Universum.andM s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+- hint: { lhs: and <$> sequence s, rhs: Universum.andM s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+
+- hint: { lhs: fmap or (sequence s), rhs: Universum.orM s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+- hint: { lhs: or <$> sequence s, rhs: Universum.orM s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+
+- hint: { lhs: fmap and (mapM f s), rhs: Universum.allM f s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+- hint: { lhs: and <$> mapM f s, rhs: Universum.allM f s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+- hint: { lhs: fmap or (mapM f s), rhs: Universum.anyM f s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+- hint: { lhs: or <$> mapM f s, rhs: Universum.anyM f s
+        , note: "Applying this hint would mean that some actions\n that were being executed previously would no longer be executed." }
+
+- warn: {lhs: whenM (not <$> x), rhs: unlessM x}
+- warn: {lhs: unlessM (not <$> x), rhs: whenM x}
+
+- warn: {lhs: either (const True) (const False), rhs: isLeft}
+- warn: {lhs: either (const False) (const True), rhs: isRight}
+
+- warn: {lhs: either id (const a), rhs: fromLeft a}
+- warn: {lhs: either (const b) id, rhs: fromRight b}
+
+- warn: {lhs: either Just (const Nothing), rhs: leftToMaybe}
+- warn: {lhs: either (const Nothing) Just, rhs: rightToMaybe}
+- warn: {lhs: maybe (Left l) Right, rhs: maybeToRight}
+- warn: {lhs: maybe (Right r) Left, rhs: maybeToLeft}
+
+- warn: {lhs: fromMaybe mempty, rhs: maybeToMonoid}
+- warn: {lhs: "m ?: mempty", rhs: maybeToMonoid m}
+
+
+- hint: {lhs: pure (), rhs: pass}
+- hint: {lhs: return (), rhs: pass}
+
+# Probably will be reduced when function equality is done:
```

Comment:
Regarding this `when*` stuff, let's give it a try, but maybe I'll want to remove these rules later.

Reply:
I think these are OK because it is very easy to infer from their names what they do.